--- a/driver/apdu/mbim.c
+++ b/driver/apdu/mbim.c
@@ -181,7 +181,7 @@ static int mbim_apdu_interface_transmit(struct euicc_ctx *ctx, uint8_t **rx, uin
     g_autoptr(GError) error = NULL;
 
     MbimMessage *request = mbim_message_ms_uicc_low_level_access_apdu_set_new(
-        mbim_priv->last_channel_id, MBIM_UICC_SECURE_MESSAGING_NONE, MBIM_UICC_CLASS_BYTE_TYPE_INTER_INDUSTRY, tx_len,
+        mbim_priv->last_channel_id, MBIM_UICC_SECURE_MESSAGING_NONE, MBIM_UICC_CLASS_BYTE_TYPE_EXTENDED, tx_len,
         tx, &error);
     if (!request) {
         fprintf(stderr, "error: creating apdu message failed: %s\n", error->message);
